search page styling

abeglova opened the pull request:
### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/4468

### Description (What does it do?)
This pr updates the styling of the search page. The designs are here:
https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=3375-49853&viewport=-15774%2C-1090%2C0.55&t=cX0nylncBbGUadPG-1&scaling=min-zoom&starting-point-node-id=2197%3A31345

Out of scope:
- resource cards
- The "learning materials" tab, which will combine videos and podcasts

### Screenshots (if appropriate):
<br>
<br>
<img width="1710" alt="Screenshot 2024-06-10 at 6 35 14 PM" src="https://github.com/mitodl/mit-open/assets/1934992/f1a24e92-c2eb-4a6d-ae43-8600d5308f61">
<br>
<br>
<img width="388" alt="Screenshot 2024-06-10 at 6 35 50 PM" src="https://github.com/mitodl/mit-open/assets/1934992/a7d188ac-481c-4b2c-8c95-368ebea41552">
<br>
<br>
<img width="388" alt="Screenshot 2024-06-10 at 6 36 01 PM" src="https://github.com/mitodl/mit-open/assets/1934992/d9575c82-b713-4b85-add0-f9550e9c9924">


### How can this be tested?
Go to http://localhost:8063/search verify that the page looks good and looks like the designs

Ferdi commented:
Looks great !

* is the facet column wider than the specs in figma ?
* In mobile, how to "apply" the selected filters and close the drawer ?   Do we need a ok/apply button ?


Can we change the order of facets as follows 

* professional / academic toggle
* Free 
* Topic (singular and open by default)
* Certificate (singular and open by default)
* Format (open by default)
* Offered by (collapsed by default)
* Department (collapsed by default)


One more thing( feel free to do a follow up PR):

For unit/subbrand  pages, the professional/academic toggle does not make much sense before each of them offers either academic or professional content. Can we remove the toggle those pages only ?

abeglova commented:
I addressed Ferdi's comments (and fixed a bug i found for small tablet sized screens)
<img width="1634" alt="Screenshot 2024-06-11 at 12 46 00 PM" src="https://github.com/mitodl/mit-open/assets/1934992/084047bb-9beb-4312-ae51-2a8fdbe78709">
<br>
<br>
<img width="382" alt="Screenshot 2024-06-11 at 12 46 15 PM" src="https://github.com/mitodl/mit-open/assets/1934992/01f0a70f-db73-406c-b4de-4b4a554ce2d5">
<br>
<br>
<img width="381" alt="Screenshot 2024-06-11 at 12 46 26 PM" src="https://github.com/mitodl/mit-open/assets/1934992/f1919276-6d2e-4bc2-b78c-077de5163e06">

abeglova commented:
I'm going to address switching the facet order and hiding the professional toggle for unit pages in a seperate pr which will also add a text search boxes to channel pages, just to make rebasing easier

abeglova commented:
Another screenshot with the "Clear All" button shown
<img width="384" alt="Screenshot 2024-06-11 at 1 04 54 PM" src="https://github.com/mitodl/mit-open/assets/1934992/4a3fc163-957e-4537-9b43-1841bdd05b03">

abeglova commented:
A screen recording of the mobile filter

https://github.com/mitodl/mit-open/assets/1934992/15eadb47-0ce8-4c35-9865-8ed892e087b4

steven-hatch commented:
I quickly mocked this up so you can see. ALL DRAWERS should have a mechanism in them so that they can be closed. This button should occupy the top visual space and relevant content should display underneath. 
<img width="625" alt="image" src="https://github.com/mitodl/mit-open/assets/155454663/5ec689dd-276a-44bf-b44a-aea44d847467">

Here, the space from top close section and filter content is 8px (looks more, but background of button is transparent). This will allow us to close panel and also clear filters. (Closing will keep filters as is and hides panel)

mbilalmughal commented:
@steven-hatch Can't we update it like this?
![image](https://github.com/mitodl/mit-open/assets/7486671/1e22e927-6b76-4d5b-b1fc-6d65e50b1852)


Since this section is not going to stack on mobile or any resolution, we should place it in one line; otherwise, there will be too much empty space at the top.

steven-hatch commented:
Looks good Bilal! Lets go with this. My version assumed that the drawer component had a dedicated area for close controls. Since we are going into this space already for course drawer, we can do the same thing here. I just want to make sure these drawers have similar look and feel in how they open and close. Thanks!

steven-hatch commented:
**Spacing**

1. Can we make sure that 24px of spacing is between the facets and results?
![image](https://github.com/mitodl/mit-open/assets/155454663/0fa05225-7b50-4a17-b1a6-3f0e6f894cae)

<br>

2. Using 12column grid...at full container width at 1272 the facet column should equal 300px in width and the results section should equal 948px. 
![image (6)](https://github.com/mitodl/mit-open/assets/155454663/af788c4d-4098-44f5-b900-2b89723d18ff)

<br>

3. Space between toggle, free, and rest of the facets should be 8px. 
![image](https://github.com/mitodl/mit-open/assets/155454663/27f8a371-be9d-4922-a6d8-44a20a49226d)

<br>

4. All facets under 'free' need to touch. 0 space in between and top facet (top left, top right) and bottom facet (bottom left, bottom right) should have rounded corners of 8px. 
![Screenshot 2024-06-11 at 2 16 25 PM](https://github.com/mitodl/mit-open/assets/155454663/fbb5a84e-3dd8-4d0e-8a16-1c9cd11f0144)

<br>

5. This might already be fixed, but can we align the sort selector with the right edge of container. Also, this element seems off. Its not balanced, maybe your release to RC will also address this. 
<img width="361" alt="Screenshot 2024-06-11 at 2 21 15 PM" src="https://github.com/mitodl/mit-open/assets/155454663/b05f09f5-1434-40a5-808a-66ea4004baa3">

<br>

6. Can we make sure that the space between cards is 8px? 
![image](https://github.com/mitodl/mit-open/assets/155454663/0805bf89-a0cc-4281-99d6-fa4ea23d96aa)

<br>

7. Can we also make sure that the space between result cards and pagination is 16px?
![image](https://github.com/mitodl/mit-open/assets/155454663/3b047639-2409-4282-b8fa-35dc72b70793)

<br>

8. Same for tab/sort and cards. This space should also be 16px. 
![image](https://github.com/mitodl/mit-open/assets/155454663/340d5c66-2794-4b13-b8da-42e4988dc14b)

steven-hatch commented:
Spacing on Facets - https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=4035%3A30313&t=3HNSgXQDzrwP6UPl-1

Please make sure that our specified spacing is set in the facets. The spacing can be seen in dev mode with link attached, or in the image (RC vs FigmaDev mode). This spacing is important because it makes the facets balanced and modern. 
![image](https://github.com/mitodl/mit-open/assets/155454663/a8eb3edf-77ec-49f5-8f3c-0bcd44cc2f87)

abeglova commented:
This is ready for re-review

Updated UI:
<img width="1655" alt="Screenshot 2024-06-12 at 4 20 00 PM" src="https://github.com/mitodl/mit-open/assets/1934992/97216965-ceff-4d0c-8bd1-ec03a0786c16">
<br>
<br>
<img width="1658" alt="Screenshot 2024-06-12 at 4 20 29 PM" src="https://github.com/mitodl/mit-open/assets/1934992/0467f31d-652c-4899-a2ce-169ab1d9c6f4">
<br>
<br>
<img width="367" alt="Screenshot 2024-06-12 at 4 20 49 PM" src="https://github.com/mitodl/mit-open/assets/1934992/c56b3bfc-76b3-45d2-af4d-123bce734ffb">
<br>
<br>
<img width="369" alt="Screenshot 2024-06-12 at 4 20 57 PM" src="https://github.com/mitodl/mit-open/assets/1934992/9a5745ca-8382-44af-ba8f-39caf9291dc9">
<br>
<br>
<img width="370" alt="Screenshot 2024-06-12 at 4 21 15 PM" src="https://github.com/mitodl/mit-open/assets/1934992/b87ed0f7-56a9-40f8-a2bf-2006e751df4b">

mbilalmughal commented:
A few issues remain, @abeglova. Please recheck the items that Steve and I suggested in the comments above.

jonkafton commented on frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:
```diff
@@ -293,17 +441,54 @@ const SearchDisplay: React.FC<SearchDisplayProps> = ({
               <FilterTitle>
-                <Typography variant="h5">Filters</Typography>
-                <TuneIcon fontSize="inherit" />
+                <Typography variant="h5">Filter</Typography>
```
`subtitle1` for this one and color `darkGray2`

jonkafton commented on frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:
```diff
@@ -293,17 +441,54 @@ const SearchDisplay: React.FC<SearchDisplayProps> = ({
-                <Typography variant="h5">Filters</Typography>
-                <TuneIcon fontSize="inherit" />
+                <Typography variant="h5">Filter</Typography>
+                <TuneIcon fontSize="medium" />
```
This replaced by the Remix Icon "equalizer-line" icon, https://remixicon.com/icon/equalizer-line

jonkafton commented on frontends/mit-open/src/page-components/SearchDisplay/ProfessionalToggle.tsx:
```diff
@@ -35,8 +35,11 @@ const StyledRiBriefcase3Line = styled(RiBriefcase3Line)`
   backgroundColor: theme.custom.colors.lightGray2,
   color: theme.custom.colors.darkGray2,
+  borderColor: theme.custom.colors.silverGrayLight,
+
```
Typography `subtitle3` and color `darkgray2` (all test that pretty much looks black is  `darkgray2`)

jonkafton replied:
Also, text change below (L92) to "View all"

abeglova replied:
I think Ferdi asked for it to be "All" instead of "View All". will make the other changes

jonkafton commented on frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:
```diff
@@ -12,9 +12,11 @@ import {
 } from "ol-components"
 
 import TuneIcon from "@mui/icons-material/Tune"
+import CloseIcon from "@mui/icons-material/Close"
```
Remix Icon for this one https://remixicon.com/icon/close-line. I think it's
```
import { RiCloseLine } from "@remixicon/react"
```

jonkafton commented on frontends/ol-components/src/components/SearchInput/SearchInput.tsx:
```diff
@@ -15,6 +18,29 @@ export interface SearchSubmissionEvent {
+  border-bottom-right-radius: 0;
+  max-width: 81%;
+  border-right: none;
+  height: 37px;
```
`theme.typography.body2` here. The designs show the placeholder with color `silverGrayDark`. Not sure about the active text, probably `darkGray2` again.

On mobile it's `theme.typography.body4`

jonkafton replied:
Also adds `padding: 8px 16px;`

<img width="599" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/f84fa7bc-e2ad-4c6d-b475-d862547509de">

jonkafton commented on frontends/ol-components/src/components/SearchInput/SearchInput.tsx:
```diff
@@ -15,6 +18,29 @@ export interface SearchSubmissionEvent {
+  height: 37px;
+`
+
+const StyledButton = styled(Button)`
```
The mobile view swaps out the search button for an icon inside the input, https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=4026-58868&m=dev

<img width="571" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/b139ef2a-6415-456c-90ed-7d41772e444e">

jonkafton replied:
Also adds 8px vertical padding to make taller

<img width="571" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/5a6c22df-6dcc-4cd7-bb0c-990e2e8b97a9">

abeglova replied:
That's an old design that hasn't been update on mobile. The mobile view is supposed to have the same search bar as desktop. The tabs in Figma are also not updated to the new design

mbilalmughal replied:
@jonkafton @abeglova  Updating mobile UI in a moment

mbilalmughal replied:
UI updated, Thanks for pointing it out 👍 

![image](https://github.com/mitodl/mit-open/assets/7486671/22a0d975-7aa9-4aed-9f04-281424a03fbc)

also @abeglova Please check the spacing of the [filter, clear all, & Close ](https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=4026-61843&t=Maq5E3Q8L8jxS6gd-1) along with the things we have earlier today 👍

jonkafton replied:
Thanks @mbilalmughal. Update received 

<img width="412" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/ee3ab7a5-ec8d-4e9e-a98e-43b2ba8929b7">

jonkafton commented:
Can we make style changes in @mitodl/course-search-utils, or will it impact other projects? The checkboxes in Figma are bigger and text smaller and things a bit more roomy. We could add overrides I suppose.

<img width="298" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/424c559e-08b5-4296-81da-1e33b4f262d8">

mbilalmughal commented:
> Can we make style changes in @mitodl/course-search-utils, or will it impact other projects? The checkboxes in Figma are bigger and text smaller and things a bit more roomy. We could add overrides I suppose.
> 
> <img alt="image" width="298" src="https://private-user-images.githubusercontent.com/939376/339466528-424c559e-08b5-4296-81da-1e33b4f262d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyOTY0MTcsIm5iZiI6MTcxODI5NjExNywicGF0aCI6Ii85MzkzNzYvMzM5NDY2NTI4LTQyNGM1NTllLTA4YjUtNDI5Ni04MWRhLTFlMzNiNGYyNjJkOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxM1QxNjI4MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kM2EzMDAyZDlkNjExYWQxZjBmYjI4OTRiZjAxZDYzMDNhMTRiN2U1ZjNjZjJjZDczMTc3NzA4NDNkZjg0YTI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QXWzfx80zZyEBQXbBlleCQwrOTqqnzvXJN1uiGiSDjg">

@jonkafton Sorry, I couldn't understand the question clearly. Are you suggesting increasing the overall weight, increase the typo and checkbox size?

steven-hatch commented:
Thanks Jon. Bilal has questions, but I think the main question you are asking is if these checkbox's are for Facets only, or will it be more of a global change. 

For now, we need a component for checkbox and radio so it will be a global change. We will have 1 version for the look and feel of checkbox's and they are needed for Facets, Lists, Profile page and maybe one other spot. 

In Figma, we will use `Checkbox` component [here](https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=1074%3A29323&t=SAEhxqUfZxIG1htI-1). There are some other uses of our Studion checkbox that will need to be swapped out in Figma so you might see some of the older styling in modals and other elements. I thought this was cleaned up, but I will do a pass after lunch to make sure.

jonkafton commented:
> @jonkafton Sorry, I couldn't understand the question clearly. Are you suggesting increasing the overall weight, increase the typo and checkbox size?

@mbilalmughal no, that was just a question for @abeglova to understand whether we can make changes to these filters to match your designs, as they live in another code repo.

abeglova commented:
I increased the check boxes!

abeglova commented:
This is ready for another round of review. New screenshots:
<img width="1576" alt="Screenshot 2024-06-13 at 3 27 34 PM" src="https://github.com/mitodl/mit-open/assets/1934992/481c94d6-8dd6-47ff-b5f5-9099f5bcc0c6">
<br>
<br>
<img width="1654" alt="Screenshot 2024-06-13 at 6 04 49 PM" src="https://github.com/mitodl/mit-open/assets/1934992/3041c1c6-96bd-46e0-9ef9-e865b0255598">
<br>
<br>
<img width="1657" alt="Screenshot 2024-06-13 at 6 05 37 PM" src="https://github.com/mitodl/mit-open/assets/1934992/dd1ffdd3-cf71-424b-be66-12ab57dd4601">
<br>
<br>
<img width="347" alt="Screenshot 2024-06-13 at 6 05 54 PM" src="https://github.com/mitodl/mit-open/assets/1934992/1b12bb9a-917b-44d9-b595-b66107cf3283">
<br>
<br>
<img width="349" alt="Screenshot 2024-06-13 at 6 06 05 PM" src="https://github.com/mitodl/mit-open/assets/1934992/94c9a668-28c3-461c-b980-9307aa65f880">
<br>
<br>
<img width="346" alt="Screenshot 2024-06-13 at 6 06 32 PM" src="https://github.com/mitodl/mit-open/assets/1934992/1b4a4da4-664e-46ab-b001-6b66d6136027">

mbilalmughal commented:
@abeglova Thanks for the updates. I have tried to mention all the needed fixes in the attached image. There are a total of 12 issues. After addressing these, please merge the PR. We'll do a second round of design QA if anything was missed, as discussed. 👍 
CC: @steven-hatch 

![UI-Fixes](https://github.com/mitodl/mit-open/assets/7486671/7b99d792-263c-4954-a809-25363d75ba62)

abeglova commented:
As discussed with Steve and Bilal on slack:

For points 4 and 5 setting fixed left-right margin for the All-Academic-Professional buttons doesn't work since the whole button group needs to be as fixed width. Setting the button height to 40px and the buttons to 20% 40% 40%  All- Academic-Professional

abeglova commented:
This is ready for re-review 
<img width="1656" alt="Screenshot 2024-06-14 at 4 13 46 PM" src="https://github.com/mitodl/mit-open/assets/1934992/2039cb53-3785-464c-8cff-1841d813536a">
<img width="342" alt="Screenshot 2024-06-14 at 4 14 02 PM" src="https://github.com/mitodl/mit-open/assets/1934992/3257016f-1849-4f84-bba6-d901a7e246f6">
<img width="347" alt="Screenshot 2024-06-14 at 4 14 10 PM" src="https://github.com/mitodl/mit-open/assets/1934992/ac411ba1-bb53-4761-aae4-517c1c3d815a">

Ferdi commented:
The screenshots look great, so pls keep the review at the code level -- we can improve with follow up prs

abeglova commented:
Screenshots after rebasing over the card updates
<img width="828" alt="Screenshot 2024-06-17 at 10 36 53 AM" src="https://github.com/mitodl/mit-open/assets/1934992/e3969695-d5d4-467d-86d5-e0c250c59711">
<br>
<br>
<img width="829" alt="Screenshot 2024-06-17 at 10 37 07 AM" src="https://github.com/mitodl/mit-open/assets/1934992/c5a001c8-0809-4e04-8342-550026e65d8d">
<br>
<br>
<img width="378" alt="Screenshot 2024-06-17 at 10 38 36 AM" src="https://github.com/mitodl/mit-open/assets/1934992/894b237b-7a84-497b-b47a-75c5445e267f">
<br>
<br>
<img width="379" alt="Screenshot 2024-06-17 at 10 38 44 AM" src="https://github.com/mitodl/mit-open/assets/1934992/de81d5bd-cdea-46c1-b48d-eb1ad3e750be">
<br>
<br>
<img width="377" alt="Screenshot 2024-06-17 at 10 38 56 AM" src="https://github.com/mitodl/mit-open/assets/1934992/2f6f8ef6-d1ff-434e-9742-37a53920f260">
